Loop/Managers/WatchDataManager.swift

```diff
@@ -253,13 +253,16 @@ final class WatchDataManager: NSObject {
                 context.glucoseTrendRawValue = trend.rawValue
             }
             
+            var savedPreMealOverride: TemporaryScheduleOverride?
             if let potentialCarbEntry = potentialCarbEntry {
+                savedPreMealOverride = loopManager.settings.preMealOverride
+                loopManager.settings.preMealOverride = nil
```

Comment:
Hmm, I thought your fix in your other PR was to make LoopDataManager not consider the pre-meal override when there was a potentialCarbEntry. Shouldn't that work here as well?

Reply:
The other PR only does this via the `BolusEntryViewModel` (which effectively does this... set `loopManager.settings.preMealOverride` to nil), but the watch does not go through that code path.

Reply:
Ah, ok, I don't think I read it closely enough then. What I thought was happening was that you changed LDM to use the `potentialCarbEntry` as a flag to determine whether to use preMealOverride.  I do think that's a great idea. 

We already pass `potentialCarbEntry` to the top level `recommendBolus` method. It then calls `recommendBolusValidatingDataRecency`, which would need to change to take a flag about whether the recommendation should consider premeal.  That same flag could be passed to `recommendBolus(forPrediction: predictedGlucose)`, which is where the target range is looked up. That lookup could be modified to use something like `glucoseTargetRangeScheduleApplyingNonPreMealOverrideIfActive`, and then you'd get a recommendation without pre-meal affecting things. 

This approach would solve the issue for both watch and BolusEntryView without needing to do any temporary state changes.

Reply:
> Ah, ok, I don't think I read it closely enough then. What I thought was happening was that you changed LDM to use the potentialCarbEntry as a flag to determine whether to use preMealOverride. I do think that's a great idea.

^ This turned out to be too difficult to achieve without massive refactoring.  We discussed this, remember? That's why I went the route I did with `BolusEntryViewModel`.

Reply:
What else would need to be refactored? I do remember discussing a path similar to this, but I think I was sidetracked by thinking that the stores' idea of overrides would need to be changed, but that's not the case. Looking at the code now, I believe it's just adding that flag to those two methods, and calling a simplified version of glucoseTargetRangeScheduleApplyingOverrideIfActive (that doesn't consider pre-meal) if the flag is true.